### PR TITLE
Improve compatibility with two-notes tremolos from Finale MusicXML im…

### PR DESCRIPTION
**Pull request by AntonioBL**
…port/export

### CODE REVIEW

**Comment by AntonioBL** [comment]
This PR implements import/export of the "filled" attribute and also exports the "base length" of the tuplet. This latter element is needed by FinaleNotepad to recognize for example a 1:8 tuplet (of duration 1/2) of two 32nd notes as a tremolo such as the ones written by the "easy tremolos" FInale plugin. See https://musescore.org/en/node/90736

Probably Travis will complain about some of the tests. I will take care of them.

**Comment by lasconic** [comment]
@AntonioBL  Would you mind making an issue and reference it in the commit message? It will help in the future if we need to dig out why we made this change. Thanks !

@lvinken Not sure if you get notified about these PRs. If you are around and want to have a look, be my guest! No hurry.

**Comment by AntonioBL** [comment]
@lasconic Actually I tried to tackle two slightly different problems related to the same issue with this PR.
I created https://musescore.org/en/node/92871 to track the "filled" question.
The other issue about the "base length" is related to the import of MusicXML exported by MuseScore into FinaleNotepad: without the "normal-type" tag FinaleNotepad does not interprets the 1:8 tuplet as the tremolos exported by the "easy tremolos" Finale plugin. See https://musescore.org/en/node/90736

With this PR the "filled" and "normal-type" tags are preserved in import/export.

**Comment by lasconic** [comment]
Sure. Can you reference this issue 90736 in the commit message then. Commits are not linked any more their mother PR once merged (as far as I know)

**Comment by AntonioBL** [comment]
I found that the normal-type is already defined in the exportxml in the current code, but it is working only if the tuplet contains notes of different values.
FinaleNotepad requires it also if the ratio is not set correctly for the tuplet, in particular: a 1:8 tuplet of one 1/16 note which actually contains two 1/32 notes.

I will split the two contributions into two different commits and I will try to tackle with that particular case of the "normal-type" (and with Travis tests).

Closing this PR for the moment.

**Comment by lvinken** [comment]
Am still around and following the developments. Will have a look, cannot commit to a specific date (real life interfering).

**Comment by lvinken** [comment]
Had a look. In terms of functionality: works flawlessly, nice work !

In terms of coding, in importmxmlpass2.cpp, I would like to propose to change int noteheadFilled into a QString. In the if name == note head body, you can assign the value of the filled attribute directly to noteheadFilled and later test for the string value. This removes variable noteheadFilled, which seems to serve only to replace string values "", "yes" and "no" by magic constants -1, 1 and 0.

File: mscore/exportxml.cpp [hunk 1]
```diff
@@ -2256,6 +2256,8 @@ static int determineTupletNormalTicks(ChordRest const* const chord)
       for (int i = 1; i < t->elements().size(); ++i)
             if (t->elements().at(0)->duration().ticks() != t->elements().at(i)->duration().ticks())
                   return t->baseLen().ticks();
+      if (t->elements().size() != t->ratio().numerator())
+                  return t->baseLen().ticks();
```
Comment (lasconic) [inline mscore/exportxml.cpp]: indent is not good here

**Comment by AntonioBL** [comment]
Ok. Thank you for the comments. I will amend the PR and resubmit (probably tomorrow).

**Comment by AntonioBL** [comment]
Re-submitted and rebased :-)

**Pull request merged**